src/vision/VisionProcessor.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+import rospy
+from sensor_msgs.msg import Image, CompressedImage
+from robosub.msg import visionPos
+import cv2
+import numpy as np
+from image_converter import ToOpenCV, depthToOpenCV
+from wfov_camera_msgs.msg import WFOVImage
+
+#this function does our image processing
+#returns the location and "size" of the detected object
+def process_image(image):
+    #convert color space from BGR to HSV
+    hsv_image = cv2.cvtColor(image, cv2.COLOR_BGR2HSV)
+
+    #create bounds for our color filter
+    lower_bound = np.array([0, 10, 10])
+    upper_bound = np.array([10,255,255])
+
+    lower_bound[0] = rospy.get_param("/vision/red/min/hue")
+    lower_bound[1] = rospy.get_param("/vision/red/min/sat")
+    lower_bound[2] = rospy.get_param("/vision/red/min/val")
+    
+    upper_bound[0] = rospy.get_param("/vision/red/max/hue")
+    upper_bound[1] = rospy.get_param("/vision/red/max/sat")
+    upper_bound[2] = rospy.get_param("/vision/red/max/val")
+    
+    #execute the color filter, returns a binary black/white image
+    mask = cv2.inRange(hsv_image, lower_bound, upper_bound)
+
+    mask = cv2.morphologyEx(mask, cv2.MORPH_OPEN, np.ones((5,5), np.uint8))
+    #mask = cv2.dilate(mask, np.ones((5,5),np.uint8), iterations=3)
+
+    #display the results of the color filter
+    cv2.imshow("image_mask", mask)
+
+    #calculate the centroid of the results of the color filer
+    M = cv2.moments(mask)
+    location = None
+    magnitude = 0
+
+    imWidth = len(image)
+    imHeight = len(image[0])
```

Comment:
The python vision processing is becoming deprecated soon and I'm not sure if I should move it to the examples or just remove it.